modules/ctl/README.md

```diff
@@ -0,0 +1,197 @@
+
+The next thing we need to do is set up SSH so we can log into this machine without a password. First get the IP address for `default`.
+
+```
+$ docker-machine ip
+192.168.99.100
+$ _
+```
+
+And then copy your public key over. I will assume you already have one. The password for user `docker` on the new VM is `tcuser`. Note that you will probably need to re-do this step if you stop and re-start the machine.
+
+```
+$ cat ~/.ssh/id_rsa.pub | ssh docker@192.168.99.100 'cat >> .ssh/authorized_keys'
+The authenticity of host '192.168.99.100 (192.168.99.100)' can't be established.
+ECDSA key fingerprint is SHA256:xPMlSBTa1YWhGXGGYRiv13rkKcHrcCZGfe2Owuuln8U.
+Are you sure you want to continue connecting (yes/no)? yes
+Warning: Permanently added '192.168.99.100' (ECDSA) to the list of known hosts.
+docker@192.168.99.100's password:
+$ _
+```
+
+Verify that you can `ssh` into the VM.
+
+```
+$ ssh docker@192.168.99.100
+                        ##         .
+                  ## ## ##        ==
+               ## ## ## ## ##    ===
+           /"""""""""""""""""\___/ ===
+      ~~~ {~~ ~~~~ ~~~ ~~~~ ~~~ ~ /  ===- ~~~
+           \______ o           __/
+             \    \         __/
+              \____\_______/
+ _                 _   ____     _            _
+| |__   ___   ___ | |_|___ \ __| | ___   ___| | _____ _ __
+| '_ \ / _ \ / _ \| __| __) / _` |/ _ \ / __| |/ / _ \ '__|
+| |_) | (_) | (_) | |_ / __/ (_| | (_) | (__|   <  __/ |
+|_.__/ \___/ \___/ \__|_____\__,_|\___/ \___|_|\_\___|_|
+Boot2Docker version 17.03.0-ce, build HEAD : f11a204 - Thu Mar  2 00:14:47 UTC 2017
+Docker version 17.03.0-ce, build 3a232c8
+docker@default:~$ exit
+Connection to 192.168.99.100 closed.
+$ _
+```
+
+You can do `docker-machine stop` to shut down the VM and `docker-machiner start` to start it back up again. Anyway we're now all set for a "remote" deployment.
+
+## Deploying and Upgrading Gem
+
+Ok so `gemctl` is an alias in the build that lets you run deployment stuff from the sbt prompt. If you just type `gemctl` at the sbt prompt you'll get a help message. `gemctl -h` will list available commands, and `gemctl <command> -h` will show details on any specific command. If the help messages aren't helpful, please improve them.
+
+#### Packaging the Application
+
+In order to deply the app we must first create a **docker image** of our application. We do this via sbt-native-packager as follows:
+
+```
+> telnetd/docker:publishLocal
+```
+
+Image names have the form `<repo>/<artifact>:<version>`. Ours will be `geminihlsw/gem-telnetd:<version>` where the version is the git commit hash, or the hash followed by `-UNCOMMITTED` if there are uncommitted changes. This give us very fine-grained versioning and a reliable partial ordering to know that an upgrade is legal. Note that if the filesystem version isn't the same as the version computed by sbt you'll need to reload the project. The prompt will include a message to this effect.
```

Comment:
Do you have access to this version at the application level if e.g. you want to display it on the UI